### PR TITLE
Dag optimization

### PR DESCRIPTION
**Pull request by udpatil**
In this, I made some fixes to the completion signal maps builder, and added more testing for it. Additionally, we eliminate edges between access operations for the same tx index, helping make the graph a bit sparser.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -130,7 +130,7 @@ require (
 )
 
 replace (
-	github.com/cosmos/cosmos-sdk => github.com/sei-protocol/sei-cosmos v0.1.67
+	github.com/cosmos/cosmos-sdk => ../sei-cosmos //github.com/sei-protocol/sei-cosmos v0.1.68
```
Comment (udpatil) [inline go.mod]: self reminder to revert this prior to merging

File: app/app.go [hunk 2]
```diff
@@ -967,30 +1067,17 @@ func (app *App) ProcessBlock(ctx sdk.Context, txs [][]byte, req BlockProcessRequ
...
-				Code:      deliverTxResp.Code,
-				Data:      deliverTxResp.Data,
-				Log:       deliverTxResp.Log,
-				Info:      deliverTxResp.Info,
-				GasWanted: deliverTxResp.GasWanted,
-				GasUsed:   deliverTxResp.GasUsed,
-				Events:    deliverTxResp.Events,
-				Codespace: deliverTxResp.Codespace,
-			})
+		if err == ErrCycleInDAG {
```
Comment (philipsu522) [inline app/app.go]: we probably want to handle other errors too. i.e. 
```
tx, err := app.txDecoder(txBytes) // TODO: results in repetitive decoding for txs with runtx decode (potential optimization)
if err != nil {
	return nil, err
}
```
 in BuildDependencyDag

**Review by philipsu522 — APPROVED** [review]

**Review by codchen — APPROVED** [review]

File: app/app.go (hunk 2 above)
Comment (udpatil) [reply app/app.go]: we actually have addressed this in a previous PR that is already merged into 2.0.0beta I believe. 
https://github.com/sei-protocol/sei-chain/commit/e573b4ab96da1c1cceb461134cf2b3e7bd92ca4c#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7R972
This is currently on top of brandon's changes, and those need to be rebased on top of the merged commit that fixed this error branching.

File: go.mod (hunk 1 above)
Comment (BrandonWeng) [reply go.mod]: fyi, i bumped it to `v0.1.69`, should be in 2.0.0

**Review by BrandonWeng — APPROVED** [review]

**Pull request merged**